testing/test_start_direction.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+
+import os
+import time
+import unittest
+
+import numpy as np
+import cv2
+
+import context
+import roboplot.config as config
+import roboplot.imgproc.image_analysis as image_analysis
+
+
+class StartDirectionTest(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super(StartDirectionTest, self).__init__(*args, **kwargs)
+        self.path_to_test_data = os.path.join(config.test_data_dir, 'StartDirectionTest')
+
+    def test_North(self):
+        start_direction = self.startDirectionTest('north')
+        self.assertEqual(start_direction, image_analysis.Direction.NORTH)
+
+    def test_East(self):
+        start_direction = self.startDirectionTest('east')
+        self.assertEqual(start_direction, image_analysis.Direction.EAST)
+
+    def test_South(self):
+        start_direction = self.startDirectionTest('south')
+        self.assertEqual(start_direction, image_analysis.Direction.SOUTH)
+ 
+    def test_West(self):
+        start_direction = self.startDirectionTest('west')
+        self.assertEqual(start_direction, image_analysis.Direction.WEST)
+
+    def test_NorthNorthEast(self):
+        start_direction = self.startDirectionTest('northnortheast')
+        # Note that this is not the optimum result but the process should be able to handdle it.
+        # If not the method will need to be revisited
+        self.assertEqual(start_direction, image_analysis.Direction.EAST)
+
+    def test_EastNorthEast(self):
+        start_direction = self.startDirectionTest('eastnortheast')
+        # Note that this is not the optimum result but the process should be able to handdle it.
+        # If not the method will need to be revisited
+        self.assertEqual(start_direction, image_analysis.Direction.NORTH)
+
+    def test_CurveSouthWest(self):
+        start_direction = self.startDirectionTest('curvesouthwest')
+        self.assertEqual(start_direction, image_analysis.Direction.SOUTH)
+
+    def test_CurveWestSouth(self):
+        start_direction = self.startDirectionTest('curvewestsouth')
+        self.assertEqual(start_direction, image_analysis.Direction.WEST)
+
+    def startDirectionTest(self, filename_without_extension):
+        # Load image
+        image = cv2.imread(os.path.join(self.path_to_test_data, filename_without_extension + '.jpg'), cv2.IMREAD_GRAYSCALE)
```

Comment:
I suspect this isn't a problem for you because you are running windows, but the tests fail for me for East, North, South and West because it can't find the image.
Maybe rename these images to have lower case names? Or pass in capitalised names to `startDirectionTest`?